casadm/cas_lib.c

```diff
@@ -565,6 +591,154 @@ int get_dev_path(const char* disk, char* buf, size_t num)
 	return err;
 }
 
+/* Indicate whether given entry in /dev/disk/by-id should be ignored -
+   we ignore software created links like 'lvm-' since these can point to
+   both CAS exported object and core device depending on initialization order.
+*/
+static bool dev_link_blacklisted(const char* entry)
+{
+	static const char* const prefix_blacklist[] = {"lvm", "md-name"};
+	static const unsigned count = ARRAY_SIZE(prefix_blacklist);
+	const char* curr;
+	unsigned i;
+
+	for (i = 0; i < count; i++) {
+		curr = prefix_blacklist[i];
+		if (!strncmp(entry, curr, strnlen_s(curr, MAX_STR_LEN)))
+			return true;
+	}
+
+	return false;
+}
+
+/* Indicate whether given path should be passed without check */
+static bool is_dev_link_whitelisted(const char* path, size_t path_len)
+{
+	regex_t regex;
+	int result;
+	static const char* const whitelist[] = { "cas[:0-9:]\\+-[:0-9:]\\+" };
+	static const unsigned count = ARRAY_SIZE(whitelist);
+	char temp_path[MAX_STR_LEN];
+	size_t i;
+
+	if (strncpy_s(temp_path, MAX_STR_LEN, path, path_len))
+		return FAILURE;
+
+	for (i = 0; i < count; i++) {
+		result = regcomp(&regex, whitelist[i], 0);
+		if (result)
+			return FAILURE;
+
+		result = regexec(&regex, temp_path, 0, NULL, 0);
+		if (!result) {
+			regfree(&regex);
+			return true;
+		}
+		regfree(&regex);
+	}
+
+	return false;
+}
+
+/* check if given dev_path is whitelisted and then pass it as path or not */
+static int get_dev_link_if_whitelisted(const char* dev_path, char* path, size_t num)
+{
+	if (is_dev_link_whitelisted(dev_path, num)){
+		strncpy_s(path, num, dev_path, strnlen_s(dev_path, MAX_STR_LEN));
+		return SUCCESS;
+	} else {
+		return FAILURE;
+	}
+}
+
+/* get device link starting with /dev/disk/by-id */
+static int get_dev_link(const char* disk, char* buf, size_t num)
```

Comment:
Since we require user to provide by-id path, there is no need for this function to exist

Reply:
It's the redundant check and I can drop it, but need to separate blacklist checking. Thanks.